### ISSUE TITLE
Ubuntu Server 22.04 LTS removes dotnet-runtime-dependencies and tries to autoremove libltnng-ust1

### ISSUE THREAD

**Issue by SanglierF** [dotnet/runtime#99752]
### Description

Ubuntu doesn't think that dotnet-dependencies-are needed for aspnetcore-runtime-8.0 and dotnet-runtime-8.0.
It removes dotnet-runtime-deps-8.0 and after that it tries to autoremove liblttng-ust1 liblttng-ust-common1 liblttng-ctl5 libraries.
It broke on two servers after unattended upgrades.

### Reproduction Steps

Install aspnetcore-runtime-8.0.
After it's installed it says it wants to autoremove liblttng-ust1 liblttng-ust-common1 liblttng-ctl5.

### Expected behavior

It doesn't try to autoremove these libraries.

### Actual behavior

It autoremoves these libraries.
```
cat /var/log/dpkg.log
2024-03-13 06:54:04 upgrade aspnetcore-runtime-8.0:amd64 8.0.2-1 8.0.3-0ubuntu1~                                                                                  22.04.1
2024-03-13 06:54:04 status half-configured aspnetcore-runtime-8.0:amd64 8.0.2-1
2024-03-13 06:54:04 status unpacked aspnetcore-runtime-8.0:amd64 8.0.2-1
2024-03-13 06:54:04 status half-installed aspnetcore-runtime-8.0:amd64 8.0.2-1
2024-03-13 06:54:04 status unpacked aspnetcore-runtime-8.0:amd64 8.0.3-0ubuntu1~    
(...)
2024-03-13 06:54:11 status installed dotnet-runtime-deps-8.0:amd64 8.0.2-1
2024-03-13 06:54:11 remove dotnet-runtime-deps-8.0:amd64 8.0.2-1 <none>
2024-03-13 06:54:11 status half-configured dotnet-runtime-deps-8.0:amd64 8.0.2-1
2024-03-13 06:54:11 status half-installed dotnet-runtime-deps-8.0:amd64 8.0.2-1
2024-03-13 06:54:11 status config-files dotnet-runtime-deps-8.0:amd64 8.0.2-1
2024-03-13 06:54:11 status not-installed dotnet-runtime-deps-8.0:amd64 <none>
(...)
```

### Configuration

Ubuntu Server 22.04 LTS
aspnetcore-runtime-8.0
dotnet-runtime-8.0
dotnet-sdk-8.0

**Comment by kkOhioGuy** [comment]
I'm confirming the same on a few of my Ubuntu 22.04 servers running 8.x

**Comment by banmohan** [comment]
Yes it is happening for me too on my Ubuntu 22.04 vps running version 8.x

**Comment by dave80int** [comment]
seem to happen to me when a new version of the dotnet runner is released...

**Comment by SanglierF** [comment]
> It's crazy that this happens and I can't pin down anything that would trigger it. It happens when I haven't run any sort of updates or rebooted. I wonder if it's timed.... just happened our our 4 dn8 servers yesterday, 10/10. The quick solution is to reinstall and restart the services or reboot but it's just nuts that the packages are seemingly spontaneously removing themselves.

Maybe you have Ubuntu's unattended upgrades enabled? It did surprise me which is how I made this issue in the first place.

I don't have any workaround since we just changed all servers distro to oS Leap 15 and I manually update them from time to time.

**Comment by kkOhioGuy** [comment]
That's definitely the main suspect. I've been looking at the dpkg and apt logs and see half-installs and removals. I'm also reading that the conflict between using the Ubuntu repo's along side of MS for this can cause issues.

**Comment by am11** [comment]
cc @mirespace

**Comment by mirespace** [comment]
Thanks, @am11, for the mention! We'll look into this and share any updates as soon as possible.

**Comment by dviererbe** [comment]
Thank you for reporting this issue :)

It looks like that you are running into a migration issue from Microsoft to Canonical built .NET deb packages. One indication of this is that:
1. we (Canonical) do not build a binary package with the name `dotnet-runtime-deps-8.0` and
2. the version numbers you are upgrading from do not contain an `ubuntu…` suffix.

Unfortunately the Microsoft and Canonical built packages are incompatible and can result in the errors reported here. Because of this reason Microsoft gradually started to no longer built packages for platforms that provide their own packages.

To prevent such migration issues you have to do one of the following things:
1. Remove the MS package feed
2. Configure apt to not install dotnet packages from the MS package feed (if you depend on other packages from the MS package feed)
3. Configure apt to not install dotnet packages from the Ubuntu package feed (if you prefer the dotnet packages from the MS package feed)

See the documentation for additional context and detailed instructions how to do that: https://learn.microsoft.com/en-us/dotnet/core/install/linux-package-mixup?pivots=os-linux-ubuntu#whats-going-on

Let me know if this advice helps!

**Comment by gwr** [comment]
Sorry if this is off topic, but are the build instructions for Ubuntu 22.04 up to date?
The build apparently wants sdk 10 and I don't see any easy way to get that.
What setup are developers using on Ubuntu these days?  Thanks!

**Comment by dviererbe** [comment]
@gwr 

If you want to install .NET 10 on Ubuntu 22.04 and higher¹ you can use official builds from the [.NET previews PPA](https://documentation.ubuntu.com/ubuntu-for-developers/reference/availability/dotnet/#previews-ppa) like this:
```
sudo add-apt-repository ppa:dotnet/previews
sudo apt install dotnet10
```
> [!IMPORTANT]
> Do not install the `dotnet9` and `dotnet10` package on the same system. There is a known conflict that arises when apt tries to install both: https://github.com/dotnet/source-build/issues/5049

> The build apparently wants sdk 10 and I don't see any easy way to get that.

If you want to build .NET 10 from source you require a .NET 10 SDK. .NET depends on .NET and there is no way to get around this. You can either download a pre-built from Microsoft or use our built via the method shown above.

_¹ except for Ubuntu 22.10, because it goes EoL before the .NET 10 release_
